test/mechanics/constitutive/LocalIsotropicDamage.cpp

```diff
@@ -0,0 +1,72 @@
+#include "BoostUnitTest.h"
+#include "math/EigenCompanion.h"
+#include "mechanics/constitutive/LocalIsotropicDamage.h"
+#include "mechanics/constitutive/ModifiedMises.h"
+#include "mechanics/constitutive/LinearElastic.h"
+#include "mechanics/constitutive/damageLaws/DamageLawExponential.h"
+
+#include "math/NewtonRaphson.h"
+
+
+BOOST_AUTO_TEST_CASE(OneDimensional)
+{
+    using namespace NuTo;
+
+    // Material parameters
+    const double E = 20000;
+    const double nu = 0.2;
+    NuTo::Laws::LinearElastic<1> elasticLaw(E, nu);
+
+    const double ft = 4;
+    const double fc = 10 * ft;
+    const double gf = 0.01;
+    const double kappa0 = ft / E;
+    const double beta = ft / gf;
+    const double alpha = 1;
+
+    // Define the law using policy based design principles, hopefully applied correctly
+    Laws::LinearElastic<1> linearElastic(E, nu);
+    Constitutive::DamageLawExponential dmg(kappa0, beta, alpha);
+    Constitutive::ModifiedMises<1> strainNorm(nu, fc / ft);
+    Laws::EvolutionImplicit<1, Constitutive::ModifiedMises<1>> history(strainNorm, /*numCells=*/1, /*numIps=*/1);
+    auto localDamageLaw = Laws::CreateLocalIsotropicDamage<1>(dmg, history, linearElastic);
```

Comment:
I know that it is very convenient to write, but I think for test files that are often used as an example on how to use specific features, I personally would prefer not using the auto feature.

Reply:
I could do that. Without auto and this `Create...` method, it would look like
~~~cpp
Laws::LocalIsotropicDamageLaw<1, Constitutive::DamageLawExponential, Laws::EvolutionImplicit<1, Constitutive::ModifiedMises<1>>, Laws::LinearElastic<1>> localDamageLaw(dmg, history, linearElastic);
~~~
or
~~~cpp
using DmgLaw = ...
using History = ...
using ElasticLaw = ...
Laws::LocalIsotropicDamage<1, DmgLaw, History, ElasticLaw> localDamageLaw(...);
~~~
Significantly longer. And not that easy to change since you have to type the types twice. I would prefer the second version, if we want to avoid `auto`.